test/coverage/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required (VERSION 3.8)
+
+add_library (seqan_app_coverage INTERFACE)
+
+option (SEQAN_APP_CODE_COVERAGE "Enable coverage reporting" OFF)
+
+if (SEQAN_APP_CODE_COVERAGE AND CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    target_compile_options (seqan_app_coverage INTERFACE "--coverage" "-fprofile-arcs" "-ftest-coverage")
+    target_link_libraries (seqan_app_coverage INTERFACE "gcov")
```

Comment:
...here?

Reply:
We wouldn't need any extra `seqan_app_coverage` target

Reply:
Ok, so don't do the target, but just add stuff to the `${PROJECT_NAME}_lib` here.
Everything is linked against the lib, so it should work